android/src/main/java/bbct/android/common/activity/FilterCards.java

```diff
@@ -162,6 +164,9 @@ public boolean onOptionsItemSelected(MenuItem item) {
             this.onConfirm();
             return true;
         }
+        else if (menuId == android.R.id.home) {
+            this.getActivity().onBackPressed();
+        }
```

Comment:
Same as other comment: if this can be implemented in `MainActivity.onOptionsItemSelected()` that would be preferable.